### ISSUE TITLE
NBomber references Serilog.Sinks.File 4.1.0 and not 5.0.0

### ISSUE THREAD

**Issue by MrRickwi** [PragmaticFlow/NBomber#458]
Hi, NBomber references Serilog.Sinks.File 4.1.0. Since we are using also Serilog.Sinks.File, but 5.0.0, NBomber crashes with "Method not found" exception. NBomber beta 3.0.0 solves that problem, but there is no beta of NBomber.Http that matches NBomber 3.0.0 beta.

Could you either
- Use Serilog.Sinks.File not hard-coded, but via dependency injection / appsettings.json?
- Publish a pre-release version of NBomber.Http that matches NBomber 3.0.0 pre-release version?

Thanks

**Comment by AntyaDev** [comment]
Hi @MrRickwi 
Sure, will try to solve this.
I think I will be able to deliver this patch in NBomber v2.1.5

**Comment by MrRickwi** [comment]
Cool, this will help a lot. Any plans when this will happen?

**Comment by AntyaDev** [comment]
I will try on this week

**Comment by MrRickwi** [comment]
I've seen 2.1.5 - but it still does not work if one uses Serilog.Sinks.File v5.0.0 in own projects.

The error is still the following:

```
System.MissingMethodException : Method not found: 'Serilog.LoggerConfiguration Serilog.FileLoggerConfigurationExtensions.File(Serilog.Configuration.LoggerSinkConfiguration, System.String, Serilog.Events.LogEventLevel, System.String, System.IFormatProvider, System.Nullable`1<Int64>, Serilog.Core.LoggingLevelSwitch, Boolean, Boolean, System.Nullable`1<System.TimeSpan>, Serilog.RollingInterval, Boolean, System.Nullable`1<Int32>, System.Text.Encoding, Serilog.Sinks.File.FileLifecycleHooks)'.
   at NBomber.Infra.Logger.attachFileLogger@31(String folder, LoggerConfiguration config)
   at NBomber.Infra.Logger.create(String folder, TestInfo testInfo, FSharpOption`1 createConfig, FSharpOption`1 configPath)
   at NBomber.Infra.Dependency.create(String reportFolder, TestInfo testInfo, ApplicationType appType, NodeType nodeType, NBomberContext context)
   at NBomber.DomainServices.NBomberRunner.run(NBomberContext context)
   at NBomber.FSharp.NBomberRunner.run(NBomberContext context)
   at NBomber.CSharp.NBomberRunner.Run(NBomberContext context)
```

**Comment by isaacabraham** [comment]
This is the same issue that we raised in #410 last year. We were actually looking to try out NBomber again for a commercial project but there are still a lot of issues around this. I think you might need to put slightly stricter constraints on your nuget packages e.g. not just minimum but also maximum boundaries.

**Comment by isaacabraham** [comment]
@MrRickwi This is almost certainly a caused by a breaking change in Serilog between v4 and v5 (totally reasonable to happen between major versions) - the problem is that the only constraint on NBomber is that the Sinks File package must be at least 4.1.0.

**Comment by MrRickwi** [comment]
The problem is that we use Serilog, latest version, in our project, and the project relies on Serilog.Sinks.File v5.0.0. So even putting stricter constraints on the usage will not help - because in that case we will have two versions there.

**Comment by isaacabraham** [comment]
@MrRickwi yep. I'm thinking of people getting this runtime area and not realising why (at least three issues in this repo relate to this) - at least this would push this into NuGet to flag it earlier. Upgrading seems something that should be in lockstep with an upgrade to NBomber itself to me (otherwise, people on Serilog 4 will be stuck?

**Comment by MrRickwi** [comment]
I would be happy to see a pre-release version of NBomber.Http that works together with the pre-release version of NBomber 3 - which was out there when I startet this thread, but seems to be deleted again at the moment I'm writing this. The pre-release of NBomber 3 already used Serilog 5...

**Comment by zetomatoz** [comment]
As @MrRickwi explained, we also observed the same issue with NBomber 2.1.5. We'd love to bring NBomber to prod but this is a blocking issue.

**Comment by AntyaDev** [comment]
Hi guys,

Thanks for informing me about this issue.
I published NBomber v3.0.0-beta2 which references Serilog.Sinks.File 5.0.0
Also, I updated all plugins including the enterprise version.
